pkg/apis/pravega/v1beta1/pravegacluster_types.go

```diff
@@ -1194,6 +1206,71 @@ func (p *PravegaCluster) validateConfigMap() error {
 	return nil
 }
 
+// ValidateSegmentStoreMemorySettings checks whether the user has passed the required values for segment store memory settings.
+// The required values includes: SegmentStoreResources.limits.memory, pravegaservice.cache.size.max , -XX:MaxDirectMemorySize and -Xmx.
+// Once the required values are set, the method also checks whether the following conditions are met:
+// Total Memory > JVM Heap (-Xmx) + JVM Direct Memory (-XX:MaxDirectMemorySize)
+// JVM Direct memory > Segment Store read cache size (pravegaservice.cache.size.max).
+func (p *PravegaCluster) ValidateSegmentStoreMemorySettings() error {
+	if p.Spec.Pravega.SegmentStoreResources == nil {
+		return fmt.Errorf("Missing required value for field spec.pravega.segmentStoreResources.limits.memory")
```

Comment:
@RaulGracia Should we allow deployments without setting the resources?

Reply:
Following the offline discussion, we do need a check for `p.Spec.Pravega.SegmentStoreResources` not to be `nil`.
If we allow the `nil` object here, skip the below validation and let the defaults be assigned later by [`withDefaults()` function](https://github.com/pravega/pravega-operator/blob/b5c2920839fb0b976ba49b1f8d6b495adc373ef9/pkg/apis/pravega/v1beta1/pravega.go#L312), we cannot perform a check for user-supplied `pravegaservice.cache.size.max` value or JVM memory settings. Therefore it would make sense to make `p.Spec.Pravega.SegmentStoreResources` a required block.